fw/cache.c

```diff
@@ -328,21 +328,52 @@ tfw_cache_key_node(unsigned long key)
 	return key % num_online_nodes();
 }
 
+/**
+ * Release memory dynamically allocated
+ * for every node cpus
+ */
+static void
+tfw_release_node_cpus(void)
+{
+	int i;
+
+	for (i = 0; i < MAX_NUMNODES; i++)
+	{
+		if(!c_nodes[i].cpu) {
+			T_ERR("%s c_node deallocation interrupted on node %i\n", __func__, i);
+			break;
+		}
+		kfree(c_nodes[i].cpu);
+	}
+}
+
 /**
  * Just choose any CPU for each node to use queue_work_on() for
  * nodes scheduling. Reserve 0th CPU for other tasks.
  */
 static void
 tfw_init_node_cpus(void)
 {
-	int cpu, node;
+	int i, nr_cpus, cpu, node;
+
+	nr_cpus = num_online_cpus();
+
+	for (i = 0; i < MAX_NUMNODES; i++) {
```

Comment:
Why do we need the loops over all NUMA nodes here and in `tfw_release_node_cpus()`? Each CPU belongs to only one node, so with the nested loops we multiply CPUs by `MAX_NUMNODES`, which also comes from the Linux config and can be quite large, so we back to the same problem as in the original #2033 task